File: gallery/src/main/java/com/qkopy/gallery/helper/PreferenceHelper.java
/*
 * Created by @ajithvgiri on 4/12/18 8:03 PM
 * Copyright (c) 2018 . All rights reserved.
 * Last modified 4/12/18 7:05 PM
 */

package com.qkopy.gallery.helper;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceHelper {

    private static final String PREFS_FILE_NAME = "Gallery";

    public static void firstTimeAskingPermission(Context context, String permission, boolean isFirstTime) {
        SharedPreferences preferences = context.getSharedPreferences(PREFS_FILE_NAME, Context.MODE_PRIVATE);
        preferences.edit().putBoolean(permission, isFirstTime).apply();
    }

    public static boolean isFirstTimeAskingPermission(Context context, String permission) {
        return context.getSharedPreferences(PREFS_FILE_NAME, Context.MODE_PRIVATE).getBoolean(permission, true);
    }
}

File: gallery/src/main/java/com/qkopy/gallery/helper/PreferenceHelper.kt
package com.qkopy.gallery.helper

import android.content.Context

object PreferenceHelper {
    private const val PREFS_FILE_NAME = "Gallery"
    fun firstTimeAskingPermission(
        context: Context,
        permission: String?,
        isFirstTime: Boolean
    ) {
        val preferences = context.getSharedPreferences(
            PREFS_FILE_NAME,
            Context.MODE_PRIVATE
        )
        preferences.edit().putBoolean(permission, isFirstTime).apply()
    }

    fun isFirstTimeAskingPermission(
        context: Context,
        permission: String?
    ): Boolean {
        return context.getSharedPreferences(
            PREFS_FILE_NAME,
            Context.MODE_PRIVATE
        ).getBoolean(permission, true)
    }
}
